apps/system/lockscreen/js/lockscreen_inputpad.js

```diff
@@ -202,14 +220,18 @@
         }
 
         this.states.passCodeEntered += key;
-        this.updatePassCodeUI();
+        this.updatePassCodeUI('');
 
         if (this.states.padVibrationEnabled) {
           navigator.vibrate(this.configs.padVibrationDuration);
         }
 
         if (this.states.passCodeEntered.length === 4) {
-          this.lockScreen.checkPassCode(this.states.passCodeEntered);
+          (function(value) {
+            setTimeout(function() {
+              this.lockScreen.checkPassCode(value);
```

Comment:
I don't think that this line of code can possibly work. You're inside a nested function that you did not call .bind(this) on, so 'this' is undefined here.  If I'm right it means that this code is non-functional and it is not actually possible to unlock the phone!